From 9d47a458abe59c098ec4173ae6860fc6ce44aca7 Mon Sep 17 00:00:00 2001
From: Zhou Chang <chang.zhou@intel.com>
Date: Tue, 27 Nov 2012 13:06:15 +0800
Subject: [PATCH 10/22] Enable merrifield platform support

BZ: 68798

Enabled build script find the mk files for merrifiled.

Change-Id: I9ecbeeb281ea29f8200b8850b86b936055e0c662
[BOOT CAMP COMMON]
Signed-off-by: Zhou Chang <chang.zhou@intel.com>
Reviewed-on: http://android.intel.com:8080/76460
Reviewed-by: Liu, Shuo <shuo.liu@intel.com>
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Reviewed-by: Li, XiaojingX <xiaojingx.li@intel.com>
Tested-by: Li, XiaojingX <xiaojingx.li@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 core/config.mk |    1 +
 envsetup.sh    |    4 +++-
 2 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/core/config.mk b/core/config.mk
index 747d8ed..9da7b04 100644
--- a/core/config.mk
+++ b/core/config.mk
@@ -145,6 +145,7 @@ board_config_mk := \
 		$(SRC_TARGET_DIR)/board/$(TARGET_DEVICE)/BoardConfig.mk \
 		device/*/$(TARGET_DEVICE)/BoardConfig.mk \
 		vendor/*/$(TARGET_DEVICE)/BoardConfig.mk \
+		vendor/*/*/board/$(TARGET_DEVICE)/BoardConfig.mk \
 	))
 ifeq ($(board_config_mk),)
   $(error No config file found for TARGET_DEVICE $(TARGET_DEVICE))
diff --git a/envsetup.sh b/envsetup.sh
index c758fcc..0ec1236 100644
--- a/envsetup.sh
+++ b/envsetup.sh
@@ -1151,7 +1151,9 @@ if [ "x$SHELL" != "x/bin/bash" ]; then
 fi
 
 # Execute the contents of any vendorsetup.sh files we can find.
-for f in `/bin/ls vendor/*/vendorsetup.sh vendor/*/*/vendorsetup.sh device/*/*/vendorsetup.sh 2> /dev/null`
+#for f in `/bin/ls vendor/*/vendorsetup.sh vendor/*/*/vendorsetup.sh device/*/*/vendorsetup.sh 2> /dev/null`
+for f in `test -d device && find device -maxdepth 6 -name 'vendorsetup.sh' 2> /dev/null` \
+         `test -d vendor && find vendor -maxdepth 6 -name 'vendorsetup.sh' 2> /dev/null`
 do
     echo "including $f"
     . $f
-- 
1.7.9

